packages/trie/src/trie.ts

```diff
@@ -1055,24 +1064,24 @@ export class Trie {
 
     // update nodes
     while (stack.length) {
-      const node = stack.pop() as TrieNode
-      if (node instanceof LeafNode) {
-        key.splice(key.length - node.key().length)
-      } else if (node instanceof ExtensionNode) {
+      const node = stack.pop()
+      if (node === undefined) {
+        throw new Error('saveStack: missing node')
+      }
+      if (node instanceof LeafNode || node instanceof ExtensionNode) {
         key.splice(key.length - node.key().length)
-        if (lastRoot) {
-          node.value(lastRoot)
-        }
-      } else if (node instanceof BranchNode) {
-        if (lastRoot) {
-          const branchKey = key.pop()
-          node.setBranch(branchKey!, lastRoot)
-        }
+      }
+      if (node instanceof ExtensionNode && lastRoot !== undefined) {
+        node.value(lastRoot)
+      }
+      if (node instanceof BranchNode && lastRoot !== undefined) {
+        const branchKey = key.pop()
+        node.setBranch(branchKey!, lastRoot)
       }
       lastRoot = this._formatNode(node, stack.length === 0, opStack) as Uint8Array
     }
 
-    if (lastRoot) {
+    if (lastRoot !== undefined) {
```

Comment:
Hmm, interesting, why did this not get caught by our linter before?

Reply:
I'm not really sure, this linting sometimes behaves in ways that I don't understand, for e.g. by flagging straight booleans (e.g. isActivatedEIP) and not flagging instances like this.